tests/5.1/teams/test_target_set_num_teams.c

```diff
@@ -0,0 +1,52 @@
+//===--------------------- test_target_set_num_teams.c ----------------------===//
+//
+// OpenMP API Version 5.1 Nov 2020
+//
+// This test uses the omp_set_num_teams utility to set the default number of
+// teams allocated for team clauses. The test runs teams and checks to see if
+// the number allocated is correct and reflective of the new default set. It
+// then checks to see if an explicitly stated num_teams clause overrides
+// the omp_set_num_teams properly.
+//
+//===------------------------------------------------------------------------===//
+
+#include <omp.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include "ompvv.h"
+
+int main() {
+
+	OMPVV_TEST_OFFLOADING;
+
+	int errors = 0; 
+	int num_teams = 0;
+
+        #pragma omp target
+        {
+	  omp_set_num_teams(4);
+        }
+	
+	#pragma omp target teams map(tofrom: num_teams)
+	{
+		if (omp_get_team_num() == 0 ) {
+			num_teams = omp_get_num_teams();
```

Comment:
omp_get_max_teams() should be used instead to query the nteams ICV which is what omp_set_num_teams() is used to change.

Reply:
Given that the specification says that: 
"The omp_set_num_teams routine affects the number of threads to be used for subsequent
4 teams regions that do not specify a num_teams clause ..."
Either way could be used, I think.